hw4/FontChooser.js

```diff
@@ -0,0 +1,163 @@
+class FontChooser extends React.Component {
+  constructor (props) {
+	super(props)
+	
+	var [minSize, maxSize, fontSize] = [this.props.min, this.props.max, this.props.size] // is there a clean way to do this?
+	// ERROR HANDLING:
+	minSize = minSize <= 0 ? 1 : minSize
+	maxSize = maxSize < minSize ? minSize : maxSize
+	
+	// make sure fontSize is between min and max
+	if (fontSize <= minSize) {
+		fontSize = minSize
+	} else if (fontSize > maxSize) {
+		fontSize = maxSize
+	}
+	
+
+    this.state = {
+		timeoutID: [],
+		minSize: minSize,
+		maxSize: maxSize,
+		hideForm: true,
+		fontColor: 'black',
+		fontStyle: 'normal',
+		fontSize: fontSize,
+		fontWeight: this.props.bold ? 'bold' : 'normal' 
```

Comment:
This is directly computed from a prop, so it should just be a computed variable in your render function. A good rule to guide you here would be that props should not be entered into state.

Reply:
I'm not allowed to call `setState` inside `render()`, so I'm gonna move this kind of calculus outside of the `setState` call, near the top of the constructor.

Reply:
I meant that you shouldn't do this kind of simple math in the constructor, but I realized that this is about setting a "default" value, so it's fine.